submissions/alstep07/task-js-oop/app.js

```diff
@@ -0,0 +1,124 @@
+const FIELD_WIDTH = 505;
+const FIELD_PADDING_TOP = 50;
+const ROW_STEP = 83;
+const COL_STEP = 101;
+const START_X = 2 * COL_STEP;
+const START_Y = 4 * ROW_STEP + FIELD_PADDING_TOP;
+const MIN_X = 0;
+const MAX_X = FIELD_WIDTH - COL_STEP;
+const MIN_Y = START_Y - 5 * ROW_STEP;
+const MAX_Y = START_Y;
+const ENEMIES = 4;
+const PLAYER_PADDING = 35;
+const MIN_SPEED = 100;
+const DIFFICULTY = 200;
+
+let counter = 0;
+
+let scoreBoard = document.createElement("div");
+document.body.append(scoreBoard);
+scoreBoard.style.cssText = "padding-top: 10px; font-size: 30px; ";
+scoreBoard.textContent = `Your score: ${counter}`;
+
+const Enemy = function (x, y, speed, player) {
+	this.x = x;
+	this.y = y;
+	this.speed = speed;
+	this.player = player;
+	this.sprite = "images/enemy-bug.png";
+};
+
+const Player = function (x, y) {
+	this.x = x;
+	this.y = y;
+	this.sprite = "images/char-boy.png";
+};
+
+const player = new Player(START_X, START_Y);
+
+const allEnemies = [];
+for (let i = 0; i < ENEMIES; i++) {
+	let x = -Math.random() * FIELD_WIDTH;
+	let y = (i % 3) * ROW_STEP + FIELD_PADDING_TOP;
+	let speed = MIN_SPEED + Math.random() * DIFFICULTY;
+	let enemy = new Enemy(x, y, speed, player);
+	allEnemies.push(enemy);
+}
+
+Enemy.prototype.update = function (dt) {
+	this.x += this.speed * dt;
+	if (this.x > FIELD_WIDTH) {
+		this.x = -COL_STEP;
+	}
+	this.collision();
+};
+
+Enemy.prototype.render = function () {
+	ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.collision = function () {
```

Comment:
By convention function/method names should start with a verb as they denote action.
Nouns are reserved to denote states or objects (in general terms, not necessarily OOP objects).